### DISCUSSION TITLE
MAGIST Algorithm v0.1.0-rc.1

### DISCUSSION THREAD

**Discussion by DragonflyRobotics** [DeepShift-Labs/MAGIST-Algorithm#15] [Announcements]
## What's Changed
* Automate object detection by @DragonflyRobotics in https://github.com/DeepShift-Labs/MAGIST-Algorithm/pull/2

**Full Changelog**: https://github.com/DeepShift-Labs/MAGIST-Algorithm/commits/v0.1.0-rc.1

This is the release candidate for MAGIST. Most bugs have been resolved and the most persistent ones are being identified and resolved. Here is the summary of the most prominent features implemented in this version:

- [x] Logging system
- [x] Google scraping
  - [x] Data downloader
  - [x] Reverse Image Search
- [x] Wikipedia scraping
  - [x] Summary retrieval
  - [x] Title retrieval
- [x] Detection data management
  - [x] Image resizer
  - [x] Image integrity verification
  - [x] Image coordinate computation
  - [x] Image cropper
  - [x] Image type converter
- [x] MAGIST Lite CNN
  - [x] Entire training class(checkpoints, TensorBoard, model export, custom training loop, etc.)
  - [x] Entire prediction class(single image prediction, image batch prediction, etc.)
- [x] Image clustering and masking
- [x] Priority queueing with multithreading
- [x] Natual Language Processing
  - [x] Audio transcription(microphone and .wav file)
  - [x] Custom SelfAttention head with custom positional embedding system
- [x] NeuralDB
  - [x] NeuralDB admin utilities
  - [x] NeuralDB insert data and full-text search

Before the first stable release, here are the tasks left to complete:
- [ ] Fix final output to convert raw JSON response from NeuralDB to human-readable text.
- [ ] Adjust MAGIST Lite CNN for optimal performance.
- [ ] Fine-tune the SelfAttention head threshold for more predictable results.
- [ ] Ensure that the documentation for installation is valid for Ubuntu 20.04 LTS.

Please let me know via GitHub Issues or GitHub Discussion any concerns, suggestions, or ideas you might have. It would help us achieve our end goal of General Intelligence.

<hr /><em>This discussion was created from the release <a href='https://github.com/DeepShift-Labs/MAGIST-Algorithm/releases/tag/v0.1.0-rc.1'>MAGIST Algorithm v0.1.0-rc.1</a>.</em>
